src/DataGrid.tsx

```diff
@@ -698,13 +700,11 @@ function DataGrid<R, SR, K extends Key>(
     if (enableEditor && isCellEditable(position)) {
       const row = rows[position.rowIdx] as R;
       setSelectedPosition({ ...position, mode: 'EDIT', row, originalRow: row });
-    } else if (
-      selectedPosition.mode !== 'SELECT' ||
-      selectedPosition.idx !== position.idx ||
-      selectedPosition.rowIdx !== position.rowIdx
-    ) {
+    } else if (isSamePosition(selectedPosition, position)) {
       // Avoid re-renders if the selected cell state is the same
       // TODO: replace with a #record? https://github.com/microsoft/TypeScript/issues/39831
+      scrollToCell(position);
```

Comment:
We should scroll to the same cell again in case it is out of view
![image](https://user-images.githubusercontent.com/1452717/134739781-1cd8099b-014d-404d-a422-4ecc797998f1.png)